### PR TITLE
Use Client Package, resolve null-pointer segfaults related to multiple active queries

### PR DESCRIPTION
**Pull request by edward-coombes-nuodb**
Changes have been made to the binding.gyp file to allow for the usage of the  `NUODB_CLIENT_PACKAGE` environment variable, to allow for a different version of the underlying CDriver to be used than what is included in the install of the existing database. If this variable is not found, the `NUODB_HOME` variable will be used instead. A final default of `/opt/nuodb/` is used if neither environment variable can be found.

The driver will now throw an error if a result set expected to be valid and have values to share is pointing to null. This case arises when multiple active queries are running at the same time.

The testing utility file now serializes queries in the `insertDataArray` function, to prevent the error described above.

### CODE REVIEW

File: test/0.multipleQueries.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+'use strict';
+
+var { Driver } = require('..');
+var should = require('should');
+var async = require('async');
+var config = require('./config');
+var helper = require('./typeHelper');
+
+describe('14. typeDouble.js', function () {
```
Comment (prosenberg1) [inline test/0.multipleQueries.js:14]: Should this be renamed 0. multipleQueries.js instead of 14 . typeDouble.js?

File: test/0.multipleQueries.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  });
+
+  describe('testing multiple queries of data', function () {
+    before('create table', function (done) {
+      async.series([
+        function (callback) {
+          helper.dropTable(connection, tableName, callback);
+        },
+        function (callback) {
+          helper.createTable(connection, tableName, 'DOUBLE', callback);
```
Comment (prosenberg1) [inline test/0.multipleQueries.js:48]: Should the type here be INTEGER?

File: package-lock.json [hunk 3]
```diff
@@ -0,0 +1,4023 @@
+{
```
Comment (rshaull) [inline package-lock.json]: It is a reasonable practice to add package-lock.json to the repo, but please do it as a separate commit.

**Comment by rshaull** [comment]
Please try to separate this into two commits: one for package-lock.json, the other for everything else.

File: test/0.multipleQueries.js (hunk 1 above)
Comment (edward-coombes-nuodb) [reply test/0.multipleQueries.js]: This file was included in the commit accidentally. I was using this as a scratch pad while working. I've reset and removed the file.

File: package-lock.json (hunk 3 above)
Comment (edward-coombes-nuodb) [reply package-lock.json:1]: I reset and added package-lock.json as a separate commit

**Review by dwhite-nuodb — APPROVED** [review]
Not an issue for this particular pull request, but in general I think we should change the version number of package-json before we actually make any delivery, which in turn will change what is in the package-lock.json file.  At some point we should incorporate a version number in the nuodb-node library, but all this is out of scope for this request, which I approve as is.

File: binding.gyp [hunk 4]
```diff
@@ -45,19 +45,19 @@
         "-std=c++14"
       ],
       "variables": {
-        "nuodb_home": "<!(echo ${NUODB_HOME-\"/opt/nuodb\"})"
+        "nuodb_client_package": "<!(echo ${NUODB_CLIENT_PACKAGE:-${NUODB_HOME:-\"/opt/nuodb\"}})"
```
Comment (madscientist) [inline binding.gyp:48]: The official environment variable that is defined to hold the root directory of the client package, like NUODB_HOME holds the root directory of the server package, is `NUOCLIENT_HOME` not `NUODB_CLIENT_PACKAGE`.
